main/field.c

```diff
@@ -385,29 +385,33 @@ static const char *renderEscapedName (const char* s,
 				      const tagEntryInfo *const tag,
 				      vString* b)
 {
-	const char* base = s;
+	int unexpected_byte = 0;
 
-	for (; *s; s++)
+	if (*s == ' ' || *s == '!')
 	{
-		int c = *s;
-		if ((c > 0x00 && c <= 0x1F) || c == 0x7F)
-		{
-			verbose ("Unexpected character (0 < *c && *c < 0x20) included in a tagEntryInfo: %s\n", base);
-			verbose ("File: %s, Line: %lu, Lang: %s, Kind: %c\n",
-				 tag->inputFileName, tag->lineNumber, getLanguageName(tag->langType), tag->kind->letter);
-			verbose ("Escape the character\n");
-			break;
-		}
-		else if (c == '\\')
-			break;
-		else
-			continue;
+		/* Don't allow a leading space or exclamation mark as it conflicts with
+		 * pseudo-tags when sorting.  Anything with a lower byte value is
+		 * escaped by renderEscapedString() already. */
+		unexpected_byte = *s;
+		vStringPut (b, '\\');
```

Comment:
The escape sequence itself is at your discretion, here I merely prefixed with a backslash, but it might make more sense to output `\x20` for space and `\x21` for an exclamation mark (their hex escape sequence, like we would have `\x1f` for Unit Separator (US)).